.appveyor.yml

```diff
@@ -1,49 +1,62 @@
-image: Visual Studio 2017
+image:
+  - Ubuntu
+  - Visual Studio 2017
 version: 2.2.0-{build}
-services:
-  - postgresql101
-environment:
-  global:
-    DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true
-    DOTNET_CLI_TELEMETRY_OPTOUT: 1
-    Test__Npgsql__DefaultConnection: Host=localhost;Database=postgres;Username=postgres;Password=Password12!
-    PGUSER: postgres
-    PGPASSWORD: Password12!
-    POSTGIS_EXE: postgis-bundle-pg10x64-setup-2.4.4-1.exe
-    NoPackageAnalysis: true  # Suppresses warning about SemVer 2.0.0 version suffixes when packing
-cache:
-  - '%USERPROFILE%\.nuget\packages -> **\*.csproj'
-  - $(POSTGIS_EXE)
 clone_depth: 10
+environment:
+  DOTNET_SKIP_FIRST_TIME_EXPERIENCE: 1
+  DOTNET_CLI_TELEMETRY_OPTOUT: 1
+  POWERSHELL_TELEMETRY_OPTOUT: 1
+  PGPASSWORD: Password12! # Used by psql on Windows.
+  POSTGIS_EXE: postgis-bundle-pg10x64-setup-2.4.4-1.exe
+  NoPackageAnalysis: true  # Suppresses warning about SemVer 2.0.0 version suffixes when packing
+for:
+-
+  matrix:
+    only:
+      - image: Visual Studio 2017
+  cache:
+    - '%USERPROFILE%\.nuget\packages -> **\*.csproj'
+    - $(POSTGIS_EXE)
+  services:
+    - postgresql101
+-
+  matrix:
+    only:
+      - image: Ubuntu
+  cache:
+    - ~/.nuget/packages -> **/*.csproj
+  services:
+    - postgresql
 install:
-  - powershell .build\setup_appveyor.ps1
-  # The following can be used to install a custom version of .NET Core
-  # - ps: Invoke-WebRequest -Uri "https://raw.githubusercontent.com/dotnet/cli/master/scripts/obtain/dotnet-install.ps1" -OutFile "install-dotnet.ps1"
-  # - ps: .\install-dotnet.ps1 -Version 2.1.300-rc1-008673
-  #
-  # - ps: $blockRdp = $true; iex ((new-object net.webclient).DownloadString('https://raw.githubusercontent.com/appveyor/ci/master/scripts/enable-rdp.ps1'))
+  - pwsh: .build/install.ps1
 before_build:
-  - appveyor-retry dotnet restore -v Minimal
+  - pwsh: .build/before_build.ps1
+  - pwsh: dotnet restore -v Minimal --disable-parallel
```

Comment:
We don't for all of them...but we do for common pathing. Otherwise `cmd` is implicit in the Windows build, and all of our `/` need to change to `\`. Using `pwsh` everywhere was just for consistency.